Sources/VLCRendererDiscoverer.m

```diff
@@ -153,16 +154,37 @@ - (void)dealloc
     return [list copy];
 }
 
+- (VLCRendererItem *)discoveredItemsContainsItem:(libvlc_renderer_item_t *)item
+{
+    for (VLCRendererItem *tmpItem in _items) {
+        if (tmpItem.rendererItem == item) {
+            return tmpItem;
+        }
+    }
+    return nil;
+}
+
 #pragma mark - Handling libvlc event callbacks
 
-- (void)itemAdded:(VLCRendererItem *)item
+- (void)itemAdded:(NSValue *)item
 {
-    [_delegate rendererDiscovererItemAdded:self item:item];
+    libvlc_renderer_item_t *pointerValue = item.pointerValue;
+    VLCRendererItem *tmp = [self discoveredItemsContainsItem:pointerValue];
+
+    if (!tmp) {
+        tmp = [[VLCRendererItem alloc] initWithCItem: pointerValue];
+        [_items addObject:tmp];
+    }
```

Comment:
then you can reduce this here to 
[self addRendererItemIfNeeded:item];
and only ever call the delegate if an item was really added